cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java

```diff
@@ -0,0 +1,755 @@
+  }
+
+  /**
+   * Tries to suspend a Workflow and expect the call completed with the status.
+   */
+  private void suspendWorkflow(Id.Program program, String runId, int expectedStatusCode) throws Exception {
+    String path = String.format("apps/%s/workflows/%s/%s/suspend", program.getApplicationId(), program.getId(), runId);
+    HttpResponse response = doPost(getVersionedAPIPath(path, Constants.Gateway.API_VERSION_3_TOKEN,
+                                                       program.getNamespaceId()));
+    Assert.assertEquals(expectedStatusCode, response.getStatusLine().getStatusCode());
+  }
+
+  /**
+   * Tries to resume a Workflow and expect the call completed with the status.
+   */
+  private void resumeWorkflow(Id.Program program, String runId, int expectedStatusCode) throws Exception {
+    String path = String.format("apps/%s/workflows/%s/%s/resume", program.getApplicationId(), program.getId(), runId);
+    HttpResponse response = doPost(getVersionedAPIPath(path, Constants.Gateway.API_VERSION_3_TOKEN,
+                                                       program.getNamespaceId()));
+    Assert.assertEquals(expectedStatusCode, response.getStatusLine().getStatusCode());
+  }
+
+  private HttpResponse getWorkflowCurrentStatus(String namespace, String app, String workflow,
+                                                String runId) throws Exception {
+    String currentUrl = String.format("apps/%s/workflows/%s/%s/current", app, workflow, runId);
+    String versionedUrl = getVersionedAPIPath(currentUrl, Constants.Gateway.API_VERSION_3_TOKEN, namespace);
+    return doGet(versionedUrl);
+  }
+
+  private String createInput(String folderName) throws IOException {
+    File inputDir = tmpFolder.newFolder(folderName);
+
+    File inputFile = new File(inputDir.getPath() + "/words.txt");
+    inputFile.deleteOnExit();
+    BufferedWriter writer = new BufferedWriter(new FileWriter(inputFile));
+    try {
+      writer.write("this text has");
+      writer.newLine();
+      writer.write("two words text inside");
+    } finally {
+      writer.close();
+    }
+
+    return inputDir.getAbsolutePath();
+  }
+
+  private Long getNextScheduledRunTime(String namespace, String app, String workflow, String schedule)
+    throws Exception {
+    String nextRunTimeUrl = String.format("apps/%s/workflows/%s/nextruntime", app, workflow);
+    String versionedUrl = getVersionedAPIPath(nextRunTimeUrl, Constants.Gateway.API_VERSION_3_TOKEN, namespace);
+    HttpResponse response = doGet(versionedUrl);
+    JsonArray array = readResponse(response, JsonArray.class);
+    JsonObject wfObject = (JsonObject) array.get(0);
+    Assert.assertNotNull(wfObject);
+    String id = wfObject.get("id").getAsString();
+    Long time = wfObject.get("time").getAsLong();
+    Assert.assertTrue(id.contains(schedule));
+    return time;
+  }
+
+  private String getStatusUrl(String namespace, String appName, String schedule) throws Exception {
```

Comment:
`URL` should be all caps (by Java naming convention, common abbreviation should be all caps when use in method name).